### PR TITLE
update dlb BUILD

### PR DESCRIPTION
**Pull request by zirain**
Commit Message: Update dlb BUILD make `istio/proxy` buildable on m1 mac, cc @daixiang0 
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: contrib/dlb/source/BUILD [hunk 1]
```diff
@@ -26,9 +26,10 @@ make(
 
 envoy_cc_contrib_extension(
     name = "connection_balancer",
-    srcs = [
-        "connection_balancer_impl.cc",
-    ],
+    srcs = select({
+        "//bazel:linux": ["connection_balancer_impl.cc"],
```
Comment (htuch) [inline contrib/dlb/source/BUILD:30]: Shouldn't this be `linux_x86_64` for consistency with the rest of the BUILD file (and also the feature is x86 only)?

Comment (zirain) [reply contrib/dlb/source/BUILD:30]: yes, thanks for point out.

**Comment by zirain** [comment]
@phlax cannot rerun github job?

**Review by daixiang0 — APPROVED** [review]
LGTM, we should only build & run it at Linux x86.

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**